server/src/main/java/org/elasticsearch/index/engine/ReaderAwareRefreshListener.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+package org.elasticsearch.index.engine;
+
+import org.apache.lucene.search.ReferenceManager;
+import org.elasticsearch.common.lucene.index.ElasticsearchDirectoryReader;
+
+import java.io.IOException;
+
+public interface ReaderAwareRefreshListener extends ReferenceManager.RefreshListener {
+
+    @Override
+    default void afterRefresh(boolean didRefresh) throws IOException {}
+
+    void afterRefresh(boolean didRefresh, ElasticsearchDirectoryReader reader) throws IOException;
```

Comment:
can we add some javadocs here mentioning that the lifecycle of the reader is managed by the listener invocation?